docs/backend/guides/service-objects/validation.md

```diff
@@ -0,0 +1,1148 @@
+"dependencies".
+
+These grouping functions can be used in multiple places, and are meant for utility. They are not a hard requirement to
+implement the validator pattern, but they are helpful organizational tools.
+
+#### Final `rules.go`
+
+
+Here is the final rules file that we have:
+
+<details>
+<summary>Final `rules.go` file</summary>
+
+```go title="pkg/services/pet/rules.go"
+package pet
+
+import (
+	"github.com/gobuffalo/validate/v3"
+
+	"github.com/transcom/mymove/pkg/appcontext"
+	"github.com/transcom/mymove/pkg/models"
+	"github.com/transcom/mymove/pkg/services"
+)
+
+// checkID checks that newPet doesn't already have an ID if we're creating a Pet, or that it matches the original Pet
+// for updates.
+func checkID() petValidator {
+	return petValidatorFunc(func(_ appcontext.AppContext, newPet models.Pet, originalPet *models.Pet) error {
+		verrs := validate.NewErrors()
+
+		if originalPet == nil {
+			if newPet.ID.IsNil() {
+				return verrs
+			}
+
+			verrs.Add("ID", "ID must not be set when creating a Pet.")
+		} else {
+			if newPet.ID != originalPet.ID {
+				verrs.Add("ID", "ID for new Pet must match original Pet ID.")
+			}
+		}
+
+		return verrs
+	})
+}
+
+// checkType checks that newPet.Type is not an empty string.
+func checkType() petValidator {
+	return petValidatorFunc(func(_ appcontext.AppContext, newPet models.Pet, originalPet *models.Pet) error {
+		verrs := validate.NewErrors()
+
+		if newPet.Type == "" {
+			verrs.Add("Type", "Type of pet must be specified.")
+		}
+
+		return verrs
+	})
+}
+
+// checkName checks that a name has been input or that one is already set, and runs the string through a string checker
+// service.
+func checkName(stringChecker services.StringChecker) petValidator {
+	return petValidatorFunc(func(appCtx appcontext.AppContext, newPet models.Pet, originalPet *models.Pet) error {
+		verrs := validate.NewErrors()
+
+		if newPet.Name == "" {
+			verrs.Add("Name", "Pet name must be defined.")
+
+			return verrs
+		}
+
+		if originalPet != nil && newPet.Name == originalPet.Name {
+			return verrs
+		}
+
+		err := stringChecker.Validate(appCtx, newPet.Name)
+
+		if err != nil {
+			verrs.Add("Name", err.Error())
+		}
+
+		return verrs
+	})
+}
+
+// customerChecks are the rules that should run for actions taken by customers
+func customerChecks(stringChecker services.StringChecker) []petValidator {
+	return []petValidator{
+		checkID(),
+		checkType(),
+		checkName(stringChecker),
+	}
+}
+
+// officeChecks are the rules that should run for actions taken by office users
+func officeChecks() []petValidator {
+	return []petValidator{
+		checkID(),
+		checkType(),
+	}
+}
+```
+
+</details>
+
+## Recap
+
+So now we've written rules for what validation we expect to happen with our data, and we've set up the validation 
+function that we can call to run through those rules. Now you can move on to setting up the actual service object.
+
+## Examples
```

Comment:
**question (non blocking):** Are there supposed to be two `##Examples` headings here?

Reply:
Nope, thanks for calling it out!